usr/lib/common/mech_ssl3.c

```diff
@@ -991,28 +1007,41 @@ CK_RV ssl3_master_key_derive(STDLL_TokData_t *tokdata,
 
     rc = object_mgr_find_in_map1(tokdata, base_key, &base_key_obj, READ_LOCK);
     if (rc != CKR_OK) {
-        TRACE_ERROR("Failed to acquire key from specified handle");
+        TRACE_ERROR("Failed to acquire key from specified handle.\n");
         if (rc == CKR_OBJECT_HANDLE_INVALID)
             return CKR_KEY_HANDLE_INVALID;
         else
             return rc;
     }
-    rc = template_attribute_find(base_key_obj->template, CKA_VALUE, &attr);
-    if (rc == FALSE) {
-        TRACE_ERROR("Could not find <the_attribute_name> in the template\n");
-        rc = CKR_FUNCTION_FAILED;
+
+    rc = template_attribute_get_bool(base_key_obj->template, CKA_DERIVE, &flag);
+    if (rc != CKR_OK) {
+        TRACE_ERROR("Could not find CKA_DERIVE for the base key.\n");
+        rc = CKR_KEY_FUNCTION_NOT_PERMITTED;
         goto error;
-    } else {
-        base_key_len = attr->ulValueLen;
-        base_key_value = attr->pValue;
+    }
 
-        if (base_key_len != 48) {
-            TRACE_ERROR("The base key's length is not 48.\n");
-            rc = CKR_KEY_FUNCTION_NOT_PERMITTED;
-            goto error;
-        }
+    if (flag == FALSE) {
+        TRACE_ERROR("CKA_DERIVE is set to FALSE.\n");
+        rc = CKR_KEY_FUNCTION_NOT_PERMITTED;
+        goto error;
     }
 
+    rc = template_attribute_get_non_empty(base_key_obj->template, CKA_VALUE,
+                                          &attr);
+    if (rc != CKR_OK) {
+        TRACE_ERROR("Could not find CKA_VALUE in the template\n");
+        goto error;
+    }
+
+    base_key_len = attr->ulValueLen;
+    base_key_value = attr->pValue;
+
+    if (base_key_len != 48) {
+        TRACE_ERROR("The base key's length is not 48.\n");
+        rc = CKR_KEY_FUNCTION_NOT_PERMITTED;
+        goto error;
+    }
```

Comment:
Hm ... from where do the 48 bytes come from? Can we possibly write out more detailed info?

Reply:
I don't know...  It was that way before, I did not dig into it....